core/src/main/kotlin/org/evomaster/core/problem/rest/service/ResourceSampler.kt

```diff
@@ -0,0 +1,278 @@
+package org.evomaster.core.problem.rest.service
+
+import com.google.inject.Inject
+import org.evomaster.core.EMConfig
+import org.evomaster.core.database.DbAction
+import org.evomaster.core.database.SqlInsertBuilder
+import org.evomaster.core.problem.rest.*
+import org.evomaster.core.problem.rest.auth.AuthenticationInfo
+import org.evomaster.core.problem.rest.auth.NoAuth
+import org.evomaster.core.problem.rest.resource.RestResourceCalls
+import org.evomaster.core.problem.rest.resource.SamplerSpecification
+import org.evomaster.core.search.Action
+import org.evomaster.core.search.EvaluatedIndividual
+import org.evomaster.core.search.service.Randomness
+import org.evomaster.core.search.service.Sampler
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * resource-based sampler
+ * the sampler handles resource-based rest individual
+ */
+abstract class ResourceSampler : Sampler<RestIndividual>() {
+
+    companion object {
+        val log: Logger = LoggerFactory.getLogger(ResourceSampler::class.java)
+    }
+
+    @Inject
+    private lateinit var ssc : ResourceSampleMethodController
+
+    @Inject
+    private lateinit var rm : ResourceManageService
+
+    @Inject
+    private lateinit var dm : ResourceDepManageService
+
+    private val authentications: MutableList<AuthenticationInfo> = mutableListOf()
+
+    private val adHocInitialIndividuals: MutableList<RestIndividual> = mutableListOf()
+
+    private var sqlInsertBuilder: SqlInsertBuilder? = null
+    var existingSqlData : List<DbAction> = listOf()
+        private set
+
+    private val samplingResourceCounter : MutableMap<String , Int> = mutableMapOf()
+    private val samplingComResourceCounter : MutableMap<String , Int> = mutableMapOf()
+    private val separatorResources = "::"
```

Comment:
in these maps, add comment specifying what is the key and what is the value

Reply:
Thanks. They are redundant code and have been removed. I am checking whether there are redundant code left.